LocalLang/Evaluator.lean

```diff
@@ -1,9 +1,30 @@
 import LocalLang.AST
+import Mathlib.Control.Traversable.Basic
 
-def BinOp.eval (n₁ n₂ : ℕ) : BinOp → ℕ
-  | add => n₁ + n₂
-  | mul => n₁ * n₂
+def BinOp.eval : BinOp → ℕ → ℕ → ℕ
+  | add, x, y => x + y
+  | mul, x, y => x * y
 
--- Maybe `m ℕ` for some monad `m`?
--- TODO: implement
-def evaluate : Program → ℕ := sorry
+def Function.updateEnv (func : Function) (V : Env) (args : List ℕ) : Env :=
```

Comment:
Please add a test (i.e. example) that checks that this correctly updates values already in the map.